### PR TITLE
marshal additional device fields

### PR DESCRIPTION
**Pull request by loyaltyarm**
fix for #126

### CODE REVIEW

**Comment by groob** [comment]
The PR looks good so far. 

Would you mind also updating the MarshalDevice/UnmarshalDevice methods here https://github.com/micromdm/micromdm/blob/master/device/device.go#L55 ?

Thanks!

**Comment by loyaltyarm** [comment]
The `go install` errors referenced from the updated commit `2004a84` above are reproduced locally as:

```
$ go install
# github.com/micromdm/micromdm/device
device/device.go:78: cannot use dev.DEPProfileStatus (type DEPProfileStatus) as type string in field value
device/device.go:80: dev.DEPAssignTime undefined (type *Device has no field or method DEPAssignTime)
device/device.go:81: dev.DEPPushTime undefined (type *Device has no field or method DEPPushTime)
device/device.go:82: cannot use dev.DEPProfileAssignedDate (type time.Time) as type int64 in field value
device/device.go:83: unknown field 'DEPProfileAssignedBy' in struct literal of type deviceproto.Device
device/device.go:84: unknown field 'LastCheckin' in struct literal of type deviceproto.Device
device/device.go:116: cannot use pb.GetDepProfileStatus() (type string) as type DEPProfileStatus in assignment
device/device.go:118: dev.DEPAssignTime undefined (type *Device has no field or method DEPAssignTime)
device/device.go:119: dev.DEPPushTime undefined (type *Device has no field or method DEPPushTime)
device/device.go:120: cannot use pb.GetDepProfileAssignedDate() (type int64) as type time.Time in assignment
device/device.go:120: too many errors
```

Any thoughts?

**Comment by groob** [comment]
There's a few different reasons these are failing. I'll point out a few:

```
device/device.go:78: cannot use dev.DEPProfileStatus (type DEPProfileStatus) as type string in field value
```

In one case it's a string, in another it's a `DEPProfileStatus`. You'll have to convert one to the other. 
`DEPProfileStatus(somestring)` will do the type conversion. 
https://tour.golang.org/basics/13

```
device/device.go:83: unknown field 'DEPProfileAssignedBy' in struct literal of type deviceproto.Device
```

Looks like the casing between the proto version of the struct and the one in `device.go` are different. That's fairly common with generated code, so just carefully look at both types and make sure you're converting one field name to the other accurately. 
Example: `DepProfileAssignedBy` vs `DEPProfileAssignedBy`. There are a few errors that are caused by mismatched filed names. 

```
device/device.go:120: cannot use pb.GetDepProfileAssignedDate() (type int64) as type time.Time in assignment
```

Here, protocolbuffers use `int64` to save time, but we need to convert to time.Time. This is probably the trickiest one but there's an example here:
https://github.com/micromdm/micromdm/blob/master/command/event.go#L63
and
here https://github.com/micromdm/micromdm/blob/master/command/event.go#L79

In short: use `e.Time.UnixNano()` to convert time.Time to `int64` and 
`time.Unix(0, pb.Time).UTC()` to take back an int64 value and convert to `time.Time`.

**Comment by loyaltyarm** [comment]
Okay, I've gotten the generated code errors errors due to casing fixed and now have just data type errors that I'm learning how to work through from your other notes:

```
$ go install
# github.com/micromdm/micromdm/device
device/device.go:80: cannot use dev.DEPProfileAssignTime (type time.Time) as type int64 in field value
device/device.go:81: cannot use dev.DEPProfilePushTime (type time.Time) as type int64 in field value
device/device.go:82: cannot use dev.DEPProfileAssignedDate (type time.Time) as type int64 in field value
device/device.go:84: cannot use dev.LastCheckin (type time.Time) as type string in field value
device/device.go:116: cannot use pb.GetDepProfileStatus() (type string) as type DEPProfileStatus in assignment
device/device.go:118: cannot use pb.GetDepProfileAssignTime() (type int64) as type time.Time in assignment
device/device.go:119: cannot use pb.GetDepProfilePushTime() (type int64) as type time.Time in assignment
device/device.go:120: cannot use pb.GetDepProfileAssignedDate() (type int64) as type time.Time in assignment
device/device.go:122: cannot use pb.GetLastCheckIn() (type string) as type time.Time in assignment
```

Interesting to learn (for other Go newbies referencing this in the future) that when working with the generated code, fixing cases requires you regularly run `go generate` so that your fixes are reflected between what is generated and your defined struct (feel free to correct this comment to reflect better explanation).

I'll post back with a hopefully working build later today.

File: device/internal/deviceproto/device.proto [hunk 1]
```diff
@@ -21,5 +21,17 @@ message Device {
...
+    string color =19;
+    string asset_tag =20;
+    bool dep_device =21;
+    string dep_profile_status =22;
+    string dep_profile_uuid =23;
+    int64 dep_profile_assign_time =24;
+    int64 dep_profile_push_time =25;
+    int64 dep_profile_assigned_date =26;
+    string dep_profile_assigned_by =27;
+    string last_check_in =28;
```
Comment (groob) [inline device/internal/deviceproto/device.proto]: `last_checkin` is a time, so should be int64, no?

Comment (loyaltyarm) [inline device/internal/deviceproto/device.proto]: It should, thanks.
